graphene_django/tests/test_query.py

```diff
@@ -1126,6 +1126,44 @@ class Query(graphene.ObjectType):
     assert len(result.data["allReporters"]["edges"]) == 4
 
 
+def test_should_have_next_page(graphene_settings):
+    graphene_settings.RELAY_CONNECTION_MAX_LIMIT = 4
+    reporters = [Reporter(**kwargs) for kwargs in REPORTERS]
+    Reporter.objects.bulk_create(reporters)
+
+    class ReporterType(DjangoObjectType):
+        class Meta:
+            model = Reporter
+            interfaces = (Node,)
+
+    class Query(graphene.ObjectType):
+        all_reporters = DjangoConnectionField(ReporterType)
+
+    schema = graphene.Schema(query=Query)
+    # Need first: 4 here to trigger the `has_next_page` logic in graphql-relay
+    # See `arrayconnection.py::connection_from_list_slice`:
+    # has_next_page=isinstance(first, int) and end_offset < upper_bound
```

Comment:
IMO this is a bug in `graphql-relay`. Without `first` param, you get `has_next_page=False` even if there are more elements... Sigh.